Docs: Update merge a pull request guideline (#43273)
grafana/grafana: contribute/merge-pull-request.md

Before actually merging a pull request there's a couple of things to take into consideration.

### Status checks

> To be able to merge a pull request all of its required status checks needs to pass including a review approval.

Reviewer: ```suggestion
Before you can merge a pull request, it must have a review approval, and all the required status checks must pass.

```

### Format the pull request title

> Make sure that the pull request title is properly formatted according to `<Area>: <Summary>` and try to make the summary short and understandable for the community as a whole.

Reviewer: ```suggestion
The pull request title should be formatted according to `<Area>: <Summary>`. Keep the summary short and understandable for the community as a whole.
```

### Assign a milestone

> A milestone **should** be added to every pull request. Several things in the Grafana release process requires at least pull requests to be in a milestone, for example [generating changelog/release notes](#include-in-changelog-and-release-notes).

Reviewer: ```suggestion
We recommended that you add a milestone to every pull request. Several things in the Grafana release process require at least one pull request to be in a milestone, for example [generating changelog/release notes](#include-in-changelog-and-release-notes).
```

Author: "require at least one pull request to be in a milestone" is not true. Soon there will be a required check on pull requests that a milestone has been added so I want this to read that it's required.

> In addition this makes it easier to track what changes did go into a certain release. Without this you're basically left with going through git commits which could be a lot harder.

Reviewer: ```suggestion
This makes it easier to track what changes go into a certain release. Without this information, release managers have to go through git commits which is not an efficient process.
```
